src/sonic-bgpcfgd/app/config.py

```diff
@@ -10,19 +10,33 @@ class ConfigMgr(object):
     """ The class represents frr configuration """
     def __init__(self):
         self.current_config = None
+        self.current_config_text = None
 
     def reset(self):
         """ Reset stored config """
         self.current_config = None
+        self.current_config_text = None
 
     def update(self):
         """ Read current config from FRR """
         self.current_config = None
+        self.current_config_text = None
         ret_code, out, err = run_command(["vtysh", "-c", "show running-config"])
         if ret_code != 0:
+            # FIXME: should we throw exception here?
             log_crit("can't update running config: rc=%d out='%s' err='%s'" % (ret_code, out, err))
             return
-        self.current_config = self.to_canonical(out)
+        text = []
+        for line in out.split('\n'):
+            if line.lstrip().startswith('!'):
+                continue
+            text.append(line)
+        text += ["     "]  # Add empty line to have something to work on, if there is no text
+        self.current_config_text = text
```

Comment:
what is the difference between current_config and current_config_text? not sure I understand the motivation here.

Reply:
current_config_text = is what frr returned by 'show run' command. So raw output.
current_config = it is parsed representation of FRR configuration.

Reply:
what about rename current_config_text to current_config_raw?

Reply:
ok

Reply:
Done

Reply:
can you update?